### PR TITLE
Redo build system

### PR DESCRIPTION
**Pull request by fulldecent**
Features:

* Remove multiple copies of files
* Remove cruft

---

You need to set GitHub Pages to serve from the `docs` folder to accept this PR.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1,82 +1,123 @@
...
+// Copy everything except the special folder _
+gulp.task('copyother', function() {
+  const f = ['**', '!**/*/', '!**/_/**', '!**/_'];
+  const f2 = function(vinyl) {
+    return !vinyl.isDirectory()
+  }
+  return gulp.src(srcPath + '**/*')
+    .pipe(filter(f))
+    .pipe(filter(f2))
+//    .pipe(gulp.symlink(distPath));
```
Comment (tomhschmidt) [inline gulpfile.js]: Please remove this line if it's no longer used

File: gulpfile.js [hunk 2]
```diff
@@ -1,82 +1,123 @@
...
+  gulp.watch(srcPath, gulp.task('copyother'));
 });
 
-// tasks that run on 'gulp' command
-gulp.task('default', ['watch']);
-gulp.task('prod', function() {
-  gulp.src('source/index.html')
-  .pipe(gulp.dest('./'))
+// BrowserSync
+gulp.task('serve', gulp.series('watch', function() {
```
Comment (tomhschmidt) [inline gulpfile.js]: I think this should be gulp.parallel, not gulp.series. series has 'watch' block serving the content

File: package.json [hunk 3]
```diff
@@ -4,27 +4,29 @@
   "description": "erc721.org",
   "main": "main.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "gulp buildclean"
```
Comment (tomhschmidt) [inline package.json]: Is this supposed to be gulp clean? buildclean isn't defined. I think we just leave the error in if we don't have tests at any rate.

**Comment by fulldecent** [comment]
@tomhschmidt All updated.

And hell yes, we have tests!

**Comment by fulldecent** [comment]
This is my first gulp website. So I'm not sure if everything is idiomatic and best practice. But it works and it's simple and it does a lot.

**Comment by tomhschmidt** [comment]
Looks like Github Pages applies styling when assets are moved to the docs folder 

image

Going to revert to move back to root

**Comment by fulldecent** [comment]
@tomhschmidt I see a comment that might have been deleted from you.

**Comment by tomhschmidt** [comment]
@fulldecent Yeah, I thought there was an issue at first after it merged, but looks like it just took a bit for the GitHub Pages cache to clear. All good now! Thanks!

**Comment by fulldecent** [comment]
Scared me! Great, thank you.

**Pull request merged**